tests/Type/data/model.php

```diff
@@ -343,6 +343,16 @@ function testRelationMethods(): void
         //assertType('Illuminate\Database\Eloquent\Builder<App\User>', $query);
     });
 
+    User::withWhereHas('accounts', function (Builder $query) {
```

Comment:
I've found that it's a bad idea to type hint the query object as that causes phpstan to drop the generic information:

```suggestion
    User::withWhereHas('accounts', function ($query) {
```

Reply:
Thank you. I will check it.

Reply:
I understood that testRelationMethods is for asserting the type of $query. In fact, the test on line 457 gives PostBuilder type hint. So without type hints, $query would simply be mixed and tests would fail.
Instead, I suggest adding the following test:

```php
function testWithWhereHas()
{
    assertType('Illuminate\Database\Eloquent\Builder<App\User>', (new User)->withWhereHas('accounts', function ($query) {
        return $query->where('active', true);
    }));
}
```

This will assert that the return value is of type Illuminate\Database\Eloquent\Builder<App\User> for the return value of withWhereHas. And this test worked in my environment.

Reply:
I was talking about dropping the generic information inside the closure

Reply:
But perhaps that was a bug that has been fixed

Reply:
Thank you for information. Did you mentioned about this issue?
https://github.com/phpstan/phpstan/issues/2934

Reply:
This is exactly why we have the commented assert lines everywhere here. So this is fine here.